managers/redis/manager_client_operations.go

```diff
@@ -0,0 +1,226 @@
+	if err != nil {
+		return fmt.Errorf("createClientRedis failed: %w", err)
+	}
+
+	if err := mn.addClientToRealm(realmName, clientNew); err != nil {
+		return fmt.Errorf("addClientToRealm failed: %w", err)
+	}
+
+	return nil
+}
+
+func (mn *RedisDataManager) DeleteClient(realmName string, clientName string) error {
+	if err := mn.deleteClientRedis(realmName, clientName); err != nil {
+		return fmt.Errorf("deleteClientRedis failed: %w", err)
+	}
+	if err := mn.deleteClientFromRealm(realmName, clientName); err != nil {
+		if errors.Is(err, errors2.ErrNotFound) || errors.Is(err, errors2.ErrZeroLength) {
+			return nil
+		}
+		return fmt.Errorf("deleteClientFromRealm failed: %w", err)
+	}
+	return nil
+}
+
+func (mn *RedisDataManager) UpdateClient(realmName string, clientName string, clientNew data.Client) error {
+	// TODO(SIA) Add transaction
+	oldClient, err := mn.GetClient(realmName, clientName)
+	if err != nil {
+		return fmt.Errorf("GetClient failed: %w", err)
+	}
+	if clientNew.ID != oldClient.ID || clientNew.Name != oldClient.Name {
+		if err := mn.DeleteClient(realmName, oldClient.Name); err != nil {
+			return fmt.Errorf("DeleteClient failed: %w", err)
+		}
+		if err := mn.addClientToRealm(realmName, clientNew); err != nil {
+			return fmt.Errorf("addClientToRealm failed: %w", err)
+		}
+	}
+
+	clientBytes, err := json.Marshal(clientNew)
+	if err != nil {
+		mn.logger.Error(sf.Format("An error occurred during Marshal Client"))
+		return fmt.Errorf("json.Marshal failed: %w", err)
+	}
+	err = mn.createClientRedis(realmName, clientNew.Name, string(clientBytes))
+	if err != nil {
+		return fmt.Errorf("createClientRedis failed: %w", err)
+	}
+
+	return nil
+}
+
+func (mn *RedisDataManager) getRealmClients(realmName string) ([]data.ExtendedIdentifier, error) {
+	realmClientsKey := sf.Format(realmClientsKeyTemplate, mn.namespace, realmName)
+	realmClients, err := getObjectsListFromRedis[data.ExtendedIdentifier](mn.redisClient, mn.ctx, mn.logger, RealmClients, realmClientsKey)
+	if err != nil {
+		return nil, fmt.Errorf("getObjectsListFromRedis failed: %w", err)
+	}
+	return realmClients, nil
+}
+
+func (mn *RedisDataManager) getRealmClient(realmName string, clientName string) (*data.ExtendedIdentifier, error) {
+	realmClients, err := mn.getRealmClients(realmName)
+	if err != nil {
+		return nil, fmt.Errorf("getRealmClients failed: %w", err)
+	}
+
+	realmHasClient := false
+	var client data.ExtendedIdentifier
+	for _, rc := range realmClients {
+		if rc.Name == clientName {
+			realmHasClient = true
+			client = rc
+			break
+		}
+	}
+	if !realmHasClient {
+		mn.logger.Debug(sf.Format("Realm: \"{0}\" doesn't have Client: \"{1}\" in Redis", realmName, clientName))
+		return nil, errors2.ErrNotFound
+	}
+	return &client, nil
+}
+
+// If such a key exists, the value will be overwritten without error
+func (mn *RedisDataManager) createClientRedis(realmName string, clientName string, clientJson string) error {
```

Comment:
probably it is better to not specify techology when we work with business object in function, because we have it in a struct name unlike from private methods that perform deep communication with redis. Because it is more clear and easier to read and understand